TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -432,6 +535,58 @@ public static DataDefinition CheckVariable(Node node, StorageArea storageArea, b
             var found = foundQualified.Select(v => v.Value);
 
             var foundCount = found.Count();
+#if DOMAIN_CHECKER
+            Scopes.Scope<VariableSymbol>.MultiSymbols result = null;
+            List<Symbol[]> foundSymbolTypedPaths = null;
+            if (SymbolTableBuilder._rootSymbolTable != null)
+            {                                
+                switch (((Symbol)node.SemanticData).Kind)
+                {
+                    case Symbol.Kinds.Program:
+                    case Symbol.Kinds.Function:
+                    {
+                        ProgramSymbol prg = (ProgramSymbol) node.SemanticData;
+                        ProgramSymbol topPrg = ExpandTopProgram(prg);
+                        result = prg.ResolveReference(area.SymbolReference, true);
+                        System.Diagnostics.Debug.Assert(result != null);
+                        //Check that we found the same number of symbols
+                        System.Diagnostics.Debug.Assert(result.Count == foundCount);
+                    }
+                        break;
+                    case Symbol.Kinds.Variable:
+                    case Symbol.Kinds.Index:
+                        //Humm....
+                        //This Storage area's SemanticData is a Variable.
+                        //Thus this situation can only appears if the variable is inside a
+                        //Typedef, or Inside a Program or a Function.
+                        //But any way we have found it.
+                        VariableSymbol @var = (VariableSymbol)node.SemanticData;
+                        if (@var.HasFlag(Symbol.Flags.InsideTypdef))
+                        {
+                            System.Diagnostics.Debug.Assert(@var.HasFlag(Symbol.Flags.InsideTypdef));
+                            System.Diagnostics.Debug.Assert(@var.TopParent(Symbol.Kinds.Typedef) != null);
+                            //We looking inside a TYPEDEF.
+                            TypedefSymbol tdSym = (TypedefSymbol)@var.TopParent(Symbol.Kinds.Typedef);
+                            foundSymbolTypedPaths = new List<Symbol[]>();
+                            result = tdSym.Get(AbstractScope.SymbolReferenceToPath(area.SymbolReference), null, foundSymbolTypedPaths);
+                            System.Diagnostics.Debug.Assert(result.Count == foundCount);
+                        }
+                        else
+                        {
+                            FunctionSymbol fun = (FunctionSymbol)@var.TopParent(Symbol.Kinds.Function);
+                            ProgramSymbol prg = (ProgramSymbol)@var.TopParent(Symbol.Kinds.Program);
+                            System.Diagnostics.Debug.Assert(prg != null || fun != null);
+                            ProgramSymbol topPrg = ExpandTopProgram(prg);
```

Comment:
In fact is that prg can never be null, because we will always have a TopParent wich is a program.
What is interresting here is to see if we are inside a function. We always Expand the Top Program which always exists. The expansion is performed on the Top Program here because we are testing, but this is not really the good place for that, the better place should be after a successfull Parsing.